### PR TITLE
new: Add modules for Placement group

### PR DESCRIPTION
**Pull request by yec-akamai**
## 📝 Description

This PR includes the following change for VM placement functionalities:
1. Add new modules:
    - placement_group: accept both label and id as identifier
    - placement_group_info
    - placement_group_list

2. In the `instance` module, allow to create a new linode instance under a placement group.

## ✔️ How to Test

Run `pip install --force -r requirements.txt`  to install required dependencies. 

Also, you need to set up environment variables to test against alpha:

```
export TEST_API_CA=$PWD/cacert.pem
export TEST_API_URL=https://.../
export LINODE_TOKEN=...
```

**Integration tests:**
```
make TEST_ARGS="-v placement_group" test
make TEST_ARGS="-v placement_group_info" test
make TEST_ARGS="-v placement_group_list" test
make TEST_ARGS="-v instance_basic" test
```

**Manual Test:**

1. In a sandbox environment, i.e. dx-devenv, run the following ansible playbook
```yaml
- name: ansible_linode Sandbox Playbook
  hosts: localhost
  tasks:
    - name: Create a Linode placement group
      linode.cloud.placement_group:
        label: my-ansible-test-pg
        region: us-east
        affinity_type: anti_affinity:local
        state: present
      register: pg_created

    - name: Get a Linode placement group info
      linode.cloud.placement_group_info:
        id: '{{ pg_created.placement_group.id }}'
      register: pg

    - name: Update a Linode placement group label
      linode.cloud.placement_group:
        id: '{{ pg.placement_group.id }}'
        label: '{{ pg.placement_group.label }}-updated'
        state: present
      register: pg_updated

    - name: List Linode placement groups for the current account
      linode.cloud.placement_group_list:
        filters:
          - name: label
            values: ['{{ pg_updated.placement_group.label }}']
      register: pg_list
```

2. Observe no error is raised.

3. Create an instance under the placement group
```yaml
    - name: Create a Linode instance under a placement group
      linode.cloud.instance:
        label: 'ansible-test-pg-{{ r }}'
        type: g6-nanode-1
        region: us-east
        placement_group:
          id: '{{ pg_updated.placement_group.id }}'
          compliant_only: false
        state: present
      register: instance_pg
```

4. Observe that the instance is created successfully.

5. Clean up the resources created.
```yaml
    - name: Delete a Linode instance
      linode.cloud.instance:
        label: '{{ instance_pg.instance.label }}'
        state: absent
      register: delete_instance_pg

    - name: Delete a placement group
      linode.cloud.placement_group:
        label: '{{ pg_updated.placement_group.label }}'
        state: absent
      register: pg_deleted
```

### CODE REVIEW

File: plugins/modules/placement_group.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+            docs_url="TBD",
+            type=FieldType.dict,
+            sample=docs.result_placement_group_samples,
+        )
+    },
+)
+
+CREATE_FIELDS = {"label", "region", "affinity_type", "is_strict"}
+# Fields that can be updated on an existing placement group
+MUTABLE_FIELDS = {"label"}
```
[the remark this answers is not in the document]
Comment (yec-akamai) [reply plugins/modules/placement_group.py:76]: Makes sense! I'm wondering if we should allow user to use the PG `id` as the identifier, then they can modify label in that case?

[the remark this answers is not in the document]
Comment (lgarber-akamai) [reply plugins/modules/placement_group.py:76]: That's a good idea! I'm not sure if we have a precedent for allowing users to specify an ID but I don't think it'd hurt to implement it here 👍

File: plugins/modules/instance.py [hunk 2]
```diff
@@ -500,6 +514,11 @@
         ],
         editable=True,
     ),
+    "placement_group": SpecField(
+        type=FieldType.dict,
+        suboptions=linode_instance_placement_group_spec,
+        description=["A Placement Group to create this Linode under."],
+    ),
 }
```
Comment (lgarber-akamai) [inline plugins/modules/instance.py:522]: It looks like re-running a playbook with an instance assigned to a PG and no changes gives the following error:

```
{"changed": false, "msg": "RuntimeError: failed to update PlacementGroup: 1946 -> {'id': 1946, 'compliant_only': False}: placement_group is a non-updatable field"}
```

We may need to introduce some logic in the update function to make sure it doesn't attempt to diff when the PG is unchanged 👀

Comment (yec-akamai) [reply plugins/modules/instance.py:522]: Good catch 👍 I excluded placement_group from the update_params to fix this issue.

Comment (lgarber-akamai) [reply plugins/modules/instance.py:522]: Thanks for fixing this! Looks like there are some backend issues right now but I'll test the fix as soon as they're resolved

File: plugins/modules/instance.py [hunk 3]
```diff
@@ -1220,6 +1240,19 @@ def _update_instance(self) -> None:
...
                 )
 
+        pg = params.get("placement_group")
+        if pg is not None:
+            if pg.get("id") != self._instance.placement_group.id:
+                self.fail(
+                    "failed to update instance {0}: placement_group.id is a "
+                    "non-updatable field".format(self._instance.label)
+                )
+            if pg.get("compliant_only"):
```
Comment (yec-akamai) [inline plugins/modules/instance.py:1250]: @lgarber-akamai The value of `compliant_only` is not returned from API and we don't have a state like tf to find its old value, so I'm thinking returning a warning here to remind user that compliant_only is not updatable.

Comment (lgarber-akamai) [reply plugins/modules/instance.py:1250]: Sounds good to me!

**Review by lgarber-akamai — APPROVED** [review]
All E2E test are passing and everything works as expected in my manual testing, great work!

**Review by zliang-akamai — APPROVED** [review]
LGTM, tested and worked well. Nice work!

**Pull request merged**